scm/etc/scripts/UFS_LSM_param.py

```diff
@@ -0,0 +1,432 @@
+import copy
+import math
+import re
+
+###############################################################################
+# Global settings                                                             #
+###############################################################################
+
+#Physical constants
+earth_radius = 6371000.0 #m
+
+# For developers: set logging level to DEBUG for additional output
+#LOGLEVEL = logging.DEBUG
+LOGLEVEL = logging.INFO
+
+###############################################################################
+# Command line arguments                                                      #
+###############################################################################
+
+parser = argparse.ArgumentParser()
+group1 = parser.add_mutually_exclusive_group(required=True)
+group1.add_argument('-l', '--location',   help='longitude and latitude in degress E and N, respectively, separated by a space', nargs=2, type=float)
+group1.add_argument('-ij','--index',      help='i,j indices within the tile (if known - bypasses search for closest model point to lon/lat location)', nargs=2, type=int)
+parser.add_argument('-g', '--grid_dir',   help='directory path containing FV3 tile supergrid files', required=True)
+parser.add_argument('-t', '--tile',       help='tile of desired point (if known - bypasses tile search if present)', type=int, choices=range(1,7))
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments():
+    """Parse command line arguments"""
+    args = parser.parse_args()
+    location = args.location
+    index = args.index
+    grid_dir = args.grid_dir
+    tile = args.tile
+    
+    #validate args
+    if not os.path.exists(grid_dir):
+        message = 'The directory {0} does not exist'.format(grid_dir)
+        logging.critical(message)
+        raise Exception(message)
+    
+    if not index:
+        if not 0 <= location[0] <= 360 :
+            message = 'The longitude {0} is outside of the range {1}'.format(location[0], '[0,360]')
+            logging.critical(message)
+            raise Exception(message)
+        
+        if not -90 <= location[1] <= 90:
+            message = 'The latitude {0} is outside of the range {1}'.format(location[1], '[-90,90]')
+            logging.critical(message)
+            raise Exception(message)
+        
+    return (location, index, grid_dir, tile)
+
+def setup_logging():
+    """Sets up the logging module."""
+    logging.basicConfig(format='%(levelname)s: %(message)s', level=LOGLEVEL)
+    
+def find_tile(loc, dir):
+    """Find the FV3 tile with the given lon/lat"""
+    #returns the integer tile number
+    
+    # should be looking in the directory with supergrid data (probably "fix" directory)
+    filename_pattern = '*grid.tile*.nc'
```

Comment:
Does this work with regional runs? For those, some files have `tile1.nc` or `tile7.nc`, and some omit `.tileN` entirely from the filename (totally confusing).

Reply:
I haven't tried this, and I was unaware of the omission of .tileN occasionally. If there are some ICs somewhere for a regional run, especially missing the .tileN, could you point me to them so that I can make any necessary modifications?